### PR TITLE
Dnahovskyi/grok 11503

### PR DESCRIPTION
**Pull request by ksumag**
second try for cloudformation.yml check

### CODE REVIEW

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 1]
```diff
@@ -48,11 +48,34 @@ Parameters:
...
+    Type: String
+    Default: 'FARGATE'
+    AllowedValues:
+      - 'EC2'
+      - 'FARGATE'
+    Description: Amazon instance type for deployment
+  Ec2KeyPairName:
+    Type: String
+    Default: 'keyname'
+    Description: Name of keypair when EC2 laumch is used
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:61]:
```suggestion
    Description: Name of keypair when EC2 launch is used
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 2]
```diff
@@ -48,11 +48,34 @@ Parameters:
...
+  Ec2PublicKey:
+    Type: String
+    Default: ''
+    Description: public key to EC2 instance when EC2 laumch is used
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:65]:
```suggestion
    Description: Public key to EC2 instance when EC2 launch is used
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 3]
```diff
@@ -48,11 +48,34 @@ Parameters:
...
+  Ec2InstanceType:
+    Type: String
+    Default: 't3.medium'
+    Description: AWS Instance type
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:69]: You need to Ec2InstanceType: one for datagrok instance and one for CVM

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 4]
```diff
@@ -3091,7 +3114,7 @@ Resources:
       Memory: '2048'
       NetworkMode: awsvpc
       RequiresCompatibilities:
-      - FARGATE
+      - !Ref LaunchType
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:3117]:
```suggestion
      - FARGATE
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 5]
```diff
@@ -2507,7 +2530,7 @@ Resources:
       DeploymentController:
         Type: ECS
       DesiredCount: 1
-      LaunchType: FARGATE
+      LaunchType: !Ref LaunchType
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:2533]:
```suggestion
      LaunchType: FARGATE
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 6]
```diff
@@ -2595,7 +2618,7 @@ Resources:
       Memory: '2048'
       NetworkMode: awsvpc
       RequiresCompatibilities:
-      - FARGATE
+      - !Ref LaunchType
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:2621]:
```suggestion
      - FARGATE
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 7]
```diff
@@ -2632,7 +2655,7 @@ Resources:
       DeploymentController:
         Type: ECS
       DesiredCount: 1
-      LaunchType: FARGATE
+      LaunchType: !Ref LaunchType
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:2658]:
```suggestion
      LaunchType: FARGATE
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 8]
```diff
@@ -2720,7 +2743,7 @@ Resources:
       Memory: '2048'
       NetworkMode: awsvpc
       RequiresCompatibilities:
-      - FARGATE
+      - !Ref LaunchType
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:2746]:
```suggestion
      - FARGATE
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 9]
```diff
@@ -2755,7 +2778,7 @@ Resources:
       DeploymentController:
         Type: ECS
       DesiredCount: 1
-      LaunchType: FARGATE
+      LaunchType: !Ref LaunchType
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:2781]:
```suggestion
      LaunchType: FARGATE
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 10]
```diff
@@ -2843,7 +2866,7 @@ Resources:
       Memory: '2048'
       NetworkMode: awsvpc
       RequiresCompatibilities:
-      - FARGATE
+      - !Ref LaunchType
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:2869]:
```suggestion
      - FARGATE
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 11]
```diff
@@ -2878,7 +2901,7 @@ Resources:
       DeploymentController:
         Type: ECS
       DesiredCount: 1
-      LaunchType: FARGATE
+      LaunchType: !Ref LaunchType
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:2904]:
```suggestion
      LaunchType: FARGATE
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 12]
```diff
@@ -2966,7 +2989,7 @@ Resources:
       Memory: '2048'
       NetworkMode: awsvpc
       RequiresCompatibilities:
-      - FARGATE
+      - !Ref LaunchType
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:2992]:
```suggestion
      - FARGATE
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 13]
```diff
@@ -3003,7 +3026,7 @@ Resources:
       DeploymentController:
         Type: ECS
       DesiredCount: 1
-      LaunchType: FARGATE
+      LaunchType: !Ref LaunchType
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:3029]:
```suggestion
      LaunchType: FARGATE
```

**Review by dnillovna — CHANGES_REQUESTED** [review]

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 14]
```diff
@@ -3114,3 +3137,134 @@ Resources:
...
+          InstanceMetadataTags: enabled
+        MaintenanceOptions:
+          AutoRecovery: default
+        InstanceInitiatedShutdownBehavior: 'stop'
+        Monitoring: false 
+        EbsOptimized: true
+        BlockDeviceMappings:
+          - Encrypted: true
+            VolumeType: 'gp3'
+            VolumeSize: '50'
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:3269]:
```suggestion
            VolumeSize: '100'
```

**Review by dnillovna — CHANGES_REQUESTED** [review]

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 15]
```diff
@@ -31,15 +67,18 @@ Parameters:
   ArnDatagrokCertificate:
     Type: String
     AllowedPattern: "(arn:([^:\\n]*):([^:\\n]*):([^:\\n]*):([^:\\n]*):(([^:\\/\\n]*)[:\\/])?(.*))?"
+    Default: arn:aws:acm:us-east-1:766822877060:certificate/11a416c1-c8f3-4e30-997d-5bf142b274d8
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:70]: leftover, delete

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 16]
```diff
@@ -31,15 +67,18 @@ Parameters:
...
     Description: AWS ACM certificate ARN for Datagrok ALB. Required parameter.
   ArnCvmCertificate:
     Type: String
     AllowedPattern: "(arn:([^:\\n]*):([^:\\n]*):([^:\\n]*):([^:\\n]*):(([^:\\/\\n]*)[:\\/])?(.*))?"
+    Default: arn:aws:acm:us-east-1:766822877060:certificate/11a416c1-c8f3-4e30-997d-5bf142b274d8
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:75]: leftover, delete

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 17]
```diff
@@ -31,15 +67,18 @@ Parameters:
...
     Description: AWS ACM certificate ARN for CVM ALB. Required parameter.
   ArnDockerHubCredential:
     Type: String
     AllowedPattern: "(arn:([^:\\n]*):([^:\\n]*):([^:\\n]*):([^:\\n]*):(([^:\\/\\n]*)[:\\/])?(.*))?"
     Description: AWS Secret Manager ARN for Docker Hub secret. Required parameter.
+    Default: arn:aws:secretsmanager:us-east-1:766822877060:secret:dockerhub-TCgq3D
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:81]: leftover, delete

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 18]
```diff
@@ -48,11 +87,42 @@ Parameters:
...
+  LaunchType:
+    Type: String
+    Default: 'FARGATE'
+    AllowedValues:
+      - 'EC2'
+      - 'FARGATE'
+    Description: Amazon instance type for deployment
+  Ec2KeyPairName:
+    Type: String
+    Default: 'keyname'
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:99]: Set an empty default

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 19]
```diff
@@ -48,11 +87,42 @@ Parameters:
...
+    Description: Name of keypair when EC2 launch is used
+  Ec2PublicKey:
+    Type: String
+    Default: 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDmmfAGuTHuaVsdgxdn/kXPrPWd8rlNlEjaYfF/w/1Qz3eRkO9xYAZKDfrUW/694f3XxoDh0kgv4+/1JXSmwCJcUuc/r2eFcaxKLzAT2dwqEhV0cWsH5AxTWpbMy9Le6FCPo87IDZN6onXdZ6HnjfQkMyj1TgFbQOgCNH3qc5yQhUC5Zy2uSkCjIO1N1UiCq/ZoUbCXExwrnLav6JemPNrcErg3G/ECkE8yKG7PJEkbndJTRoxXXkMZneRWTm65i4svKS9mDmZGQtlB9uqaHlWN+nCE9XJLu9nkzRJIHE9R2S+NNSU9kjxeVWKNDqwlSKO6xliNUSSZKYo4RMbD1ByEBCejBRulY0h07nvnMNU/sFz0QD10YKUjXac28DKpOjmO3UCuHu/fxmyH2J74n44zq3aOdqq0gl8QQfy8yfA7yI2GlRvJRd01tUmnt9rtVimmBcnk8+4JByGLZPU7g02fBgIXup5peYKgL/US12B8JJ4gQ/S3/FhXEBlV0pJjFiU= ksumag@ksumag-X751SJ'
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:103]: leftover, delete

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 20]
```diff
@@ -48,11 +87,42 @@ Parameters:
...
+    Description: Public key to EC2 instance when EC2 launch is used
+  DatagrokEc2InstanceType:
+    Type: String
+    Default: 't3.medium'
+    Description: AWS Instance type for Datagrok EC2 instance
+  CvmEc2InstanceType:
+    Type: String
+    Default: 'a1.4xlarge'
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:111]: t3.xlarge

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 21]
```diff
@@ -667,6 +737,59 @@ Resources:
...
+        TargetGroupArn:
+          Ref: GrokComputeTCP5005IntTargetGroup
+      NetworkConfiguration:
+        AwsvpcConfiguration:
+          AssignPublicIp: DISABLED
+          SecurityGroups:
+          - Ref: CvmSecurityGroup
+          Subnets:
+          - Ref: DatagrokPrivateSubnet1
+          - Ref: DatagrokPrivateSubnet2
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:776]: Useless block, for EC2 instance we should use bridge network

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 22]
```diff
@@ -813,7 +936,7 @@ Resources:
         - ContainerPort: 5005
           HostPort: 5005
           Protocol: tcp
-      Cpu: '1024'
+      Cpu: '512'
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:939]: Move to the container definition and match the values from *_container_cpu variables in terraform defaults

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 23]
```diff
@@ -822,7 +945,7 @@ Resources:
       Memory: '2048'
       NetworkMode: awsvpc
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:946]: Change network mode for EC2, it should be 'bridge'

**Review by dnillovna — CHANGES_REQUESTED** [review]

File: help/develop/admin/deploy/cloudformation/cloudformation.yml (hunk 22 above)
Comment (dnillovna) [reply help/develop/admin/deploy/cloudformation/cloudformation.yml:939]: And leave it ad it was for FARGATE

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 24]
```diff
@@ -667,6 +737,46 @@ Resources:
...
+        ContainerPort: 5005
+        TargetGroupArn:
+          Ref: GrokComputeTCP5005IntTargetGroup
+      PropagateTags: SERVICE
+      SchedulingStrategy: REPLICA
+      Tags:
+      - Key: Name
+        Value: CVM Grok Compute Service
+      - Key: datagrok.component
+        Value: datagrok
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:775]:
```suggestion
        Value: cvm
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml (hunk 22 above)
Comment (dnillovna) [reply help/develop/admin/deploy/cloudformation/cloudformation.yml:993]:
```suggestion
      Cpu: '1024'
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 25]
```diff
@@ -765,6 +883,65 @@ Resources:
       - Key: stickiness.lb_cookie.duration_seconds
         Value: 86400
     Type: AWS::ElasticLoadBalancingV2::TargetGroup
+  GrokComputeTaskDefinitionEc2:
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:886]: Why do you need a copy of TaskDefinition?

Comment (ksumag) [reply help/develop/admin/deploy/cloudformation/cloudformation.yml]: done

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 26]
```diff
@@ -1,45 +1,84 @@
...
   ArnCvmCertificate:
     Type: String
     AllowedPattern: "(arn:([^:\\n]*):([^:\\n]*):([^:\\n]*):([^:\\n]*):(([^:\\/\\n]*)[:\\/])?(.*))?"
+    Default: ' '
     Description: AWS ACM certificate ARN for CVM ALB. Required parameter.
   ArnDockerHubCredential:
     Type: String
     AllowedPattern: "(arn:([^:\\n]*):([^:\\n]*):([^:\\n]*):([^:\\n]*):(([^:\\/\\n]*)[:\\/])?(.*))?"
     Description: AWS Secret Manager ARN for Docker Hub secret. Required parameter.
+    Default: ' '
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:81]:
```suggestion
    Default: ''
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 27]
```diff
@@ -1,45 +1,84 @@
...
   H2oVersion:
     Type: String
     Default: latest
-    AllowedPattern: "([0-9]+\\.[0-9]+\\.[0-9]+)|latest"
+    AllowedPattern: "([0-9]+\\.[0-9]+\\.[0-9]+)|latest|bleeding-edge"
     Description: Enter Datagrok H2O docker image version. Default is 'latest'.
   ArnDatagrokCertificate:
     Type: String
     AllowedPattern: "(arn:([^:\\n]*):([^:\\n]*):([^:\\n]*):([^:\\n]*):(([^:\\/\\n]*)[:\\/])?(.*))?"
+    Default: ' '
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:70]:
```suggestion
    Default: ''
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 28]
```diff
@@ -1,45 +1,84 @@
...
     Description: AWS ACM certificate ARN for Datagrok ALB. Required parameter.
   ArnCvmCertificate:
     Type: String
     AllowedPattern: "(arn:([^:\\n]*):([^:\\n]*):([^:\\n]*):([^:\\n]*):(([^:\\/\\n]*)[:\\/])?(.*))?"
+    Default: ' '
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:75]:
```suggestion
    Default: ''
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 29]
```diff
@@ -48,11 +87,42 @@ Parameters:
...
+  LaunchType:
+    Type: String
+    Default: 'FARGATE'
+    AllowedValues:
+      - 'EC2'
+      - 'FARGATE'
+    Description: Amazon instance type for deployment
+  Ec2KeyPairName:
+    Type: String
+    Default: ' '
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:99]:
```suggestion
    Default: ''
```

File: help/develop/admin/deploy/cloudformation/cloudformation.yml [hunk 30]
```diff
@@ -48,11 +87,42 @@ Parameters:
...
+    Description: Name of keypair when EC2 launch is used
+  Ec2PublicKey:
+    Type: String
+    Default: ' '
```
Comment (dnillovna) [inline help/develop/admin/deploy/cloudformation/cloudformation.yml:103]:
```suggestion
    Default: ''
```

**Review by dnillovna — APPROVED** [review]

**Pull request merged**